commerce-price-list-service/src/main/java/com/liferay/commerce/price/list/service/impl/CommercePriceEntryServiceImpl.java

```diff
@@ -119,70 +143,52 @@ public CommercePriceEntry fetchCommercePriceEntry(long commercePriceEntryId)
 			OrderByComparator<CommercePriceEntry> orderByComparator)
 		throws PortalException {
 
-		PortalPermissionUtil.check(
-			getPermissionChecker(),
+		_checkCommerceCatalogPermissionByCommercePriceListId(
+			commercePriceListId,
 			CommercePriceListActionKeys.MANAGE_COMMERCE_PRICE_LISTS);
 
 		return commercePriceEntryLocalService.getCommercePriceEntries(
 			commercePriceListId, start, end, orderByComparator);
 	}
 
-	@Override
-	public List<CommercePriceEntry> getCommercePriceEntriesByCompanyId(
-			long companyId, int start, int end)
-		throws PortalException {
-
-		PortalPermissionUtil.check(
-			getPermissionChecker(),
-			CommercePriceListActionKeys.MANAGE_COMMERCE_PRICE_LISTS);
-
-		return commercePriceEntryLocalService.
-			getCommercePriceEntriesByCompanyId(companyId, start, end);
-	}
-
 	@Override
 	public int getCommercePriceEntriesCount(long commercePriceListId)
 		throws PortalException {
 
-		PortalPermissionUtil.check(
-			getPermissionChecker(),
+		_checkCommerceCatalogPermissionByCommercePriceListId(
+			commercePriceListId,
 			CommercePriceListActionKeys.MANAGE_COMMERCE_PRICE_LISTS);
 
 		return commercePriceEntryLocalService.getCommercePriceEntriesCount(
 			commercePriceListId);
 	}
 
-	@Override
-	public int getCommercePriceEntriesCountByCompanyId(long companyId)
-		throws PortalException {
-
-		PortalPermissionUtil.check(
-			getPermissionChecker(),
-			CommercePriceListActionKeys.MANAGE_COMMERCE_PRICE_LISTS);
-
-		return commercePriceEntryLocalService.
-			getCommercePriceEntriesCountByCompanyId(companyId);
-	}
-
 	@Override
 	public CommercePriceEntry getCommercePriceEntry(long commercePriceEntryId)
 		throws PortalException {
 
-		PortalPermissionUtil.check(
-			getPermissionChecker(),
+		CommercePriceEntry commercePriceEntry =
+			commercePriceEntryLocalService.getCommercePriceEntry(
+				commercePriceEntryId);
+
+		_checkCommerceCatalogPermissionByCommercePriceListId(
+			commercePriceEntry.getCommercePriceListId(),
 			CommercePriceListActionKeys.MANAGE_COMMERCE_PRICE_LISTS);
 
-		return commercePriceEntryLocalService.getCommercePriceEntry(
-			commercePriceEntryId);
+		return commercePriceEntry;
 	}
 
 	@Override
 	public List<CommercePriceEntry> getInstanceCommercePriceEntries(
 			long cpInstanceId, int start, int end)
 		throws PortalException {
 
-		PortalPermissionUtil.check(
-			getPermissionChecker(),
+		CPInstance cpInstance = _cpInstanceService.getCPInstance(cpInstanceId);
+
+		CPDefinition cpDefinition = cpInstance.getCPDefinition();
```

Comment:
Why get the `CPDefinition` here? `cpInstance.getGroupId()` is the same as `cpDefinition.getGroupId()`. Just getting the groupId from the instance would save us an unnecessary service call.